Sofa/framework/Core/src/sofa/core/collision/NarrowPhaseDetection.h

```diff
@@ -51,7 +51,20 @@ class SOFA_CORE_API NarrowPhaseDetection : virtual public Detection
 
 protected:
     /// Destructor
-    ~NarrowPhaseDetection() override { }
+    ~NarrowPhaseDetection() override {
+        for (DetectionOutputMap::iterator it = m_outputsMap.begin(); it != m_outputsMap.end(); it++)
+        {
+            DetectionOutputVector* do_vec = (it->second);
+
+            if (do_vec != nullptr)
+            {
+                do_vec->clear();
+                do_vec->release();
+            }
+            m_outputsMap.erase(it);
```

Comment:
Warning here. You erase elements of the map while iterating. To check if it is a valid operation.
Since the map is cleared anyway, I recommend to drop this operation.

Reply:
I also suggest to move the destructor definition in the cpp file